modin/backends/pandas/query_compiler.py

```diff
@@ -213,6 +213,37 @@ def caller(df, *args, **kwargs):
     return caller
 
 
+def handle_readonly_buffer(func: callable, display_name: str = None):
+    """
+    Build a function that handles read-only buffers in the workers.
+
+    This utility is for operators that have in-place side effects in pandas,
+    but are not themselves inplace.
+
+    Parameters
+    ----------
+    func : callable(pandas.DataFrame)
+        The function, which should not update the dataframe inplace.
+    display_name : str, optional
+        The function's name, which can be displayed by a progress bar.
+
+    Returns
+    -------
+    callable(pandas.DataFrame)
+        A callable function to be applied in the partitions.
+    """
+
+    def caller(df, *args, **kwargs):
+        try:
+            return func(df, *args, **kwargs)
```

Comment:
can we have a lookup table (or some other pre-defined list) of "known bad" functions to run them over a copy without running over a non-copy (for performance reasons)?

Reply:
I think it's a difficult trade-off. If we try the function first, we can avoid copying the dataframe and blowing up worker memory.

Reply:
That is true if functions fail fast when they fail... but we should use the benchmarks to tell us that.
Or we can memorize the "bad functions" on the fly, so non-first calls to them would go straight to a "copy mode".

Reply:
There might be another way to check for it, I'll see if I can find one.